In [1]:
from sklearn import datasets
from sklearn.metrics import mean_absolute_error
from sklearn.model_selection import cross_val_score
from sklearn.preprocessing import PolynomialFeatures
from sklearn.pipeline import make_pipeline
from sklearn.linear_model import LinearRegression
from sklearn import linear_model
import pandas as pd

In [3]:
#Loading Diabetes Dataset
X, y = datasets.load_diabetes(as_frame=True, scaled=False, return_X_y=True)


PERFORMING CROSS VALIDATION ON NINE POLYNOMIAL MODELS RANGING FROM 0 TO 8

In [10]:
degrees = range(9)
cross_val = []
r2_score = []
mapes = []
for degree in degrees:
    model = make_pipeline(PolynomialFeatures(degree),LinearRegression())
    scores = cross_val_score(model,X,y,cv=5,scoring='neg_mean_absolute_error').mean()
    r2_scores = cross_val_score(model,X,y,cv=5,scoring='r2').mean()
    mape_scores = cross_val_score(model,X,y,cv=5,scoring='neg_mean_absolute_percentage_error').mean()
    cross_val.append(-scores)
    r2_score.append(r2_scores)
    mapes.append(-mape_scores)

Table summarizing the cross-validation results.

In [13]:
#Summarizing the results
summary = pd.DataFrame({'degree':degrees,'cross_val':cross_val,'r2_score':r2_score,'mape':mapes})
summary

degree    cross_val      r2_score       mape
0       0    66.045624     -0.027506   0.623622
1       1    44.276499      0.482316   0.394860
2       2    69.864481     -0.957090   0.609146
3       3   330.432387   -199.795789   2.261685
4       4   657.260477   -571.083108   5.232018
5       5   562.993636   -436.856887   4.570284
6       6   742.721998  -1695.874798   6.186852
7       7  1032.681725  -5530.894075   8.666399
8       8  1475.658532 -16076.255119  12.474617

Identification of the Best Model :
From The scores of data it is clear that for degree 1 the model exhibits best results with value of r square closest to 1 then any other model and minimize the loss i.e. MAE and MAPE. 

Since R-squared measures the proportion of variance in the model, a higher R-squared value indicates a larger proportion of variability that the model could explain which means better performance.
Therefore the model has the best accuracy and the model is approx 48% better at predicting values then mean then other models betweeon degree 0 to 8.